src/core/server/saved_objects/import/regenerate_ids.ts

```diff
@@ -42,3 +43,40 @@ export const regenerateIds = (objects: SavedObject[]) => {
   }, new Map<string, { id: string; omitOriginId?: boolean }>());
   return importIdMap;
 };
+
+export const regenerateIdsWithReference = async (props: {
+  savedObjects: SavedObject[];
+  savedObjectsClient: SavedObjectsClientContract;
+  workspaces?: string[];
+  objectLimit: number;
+  importIdMap: Map<string, { id?: string; omitOriginId?: boolean }>;
+}): Promise<Map<string, { id?: string; omitOriginId?: boolean }>> => {
+  const { savedObjects, savedObjectsClient, workspaces, importIdMap } = props;
+  if (!workspaces || !workspaces.length) {
+    return savedObjects.reduce((acc, object) => {
+      return acc.set(`${object.type}:${object.id}`, { id: object.id, omitOriginId: false });
```

Comment:
I saw the existing regenerateIds function will generate new id using uuidv4, why we want to change this logic when workspace is not available?